KIPs/kip-81.md

```diff
@@ -0,0 +1,335 @@
+  function createTracker(uint256 _trackEnd) public returns (uint trackerId) {
+    // - Create a Tracker struct
+    // - Populate nodeIds and stakingToNodeId from the AddressBook.
+    //   Only those who staked at least MIN_STAKE are recorded. 
+    // - Populate nodeBalances, votingPowers, and totalVotes from
+    //   the KLAY balances of CnStakingV2 contracts
+    // - Populate nodeIdToVoter and voterToNodeId from
+    //   the last Tracker (last element of allTrackers), if any.
+  }
+
+  // Called by CnStakingV2 whenever its balance changes.
+  function notifyStake() public {
+    for (uint i=0; i < liveTrackers.length; i++) {
+      if (block.number >= trackers[i].trackEnd) {
+        deleteArrayElem(liveTrackers, i);
+        continue;
+      }
+      address nodeId = trackers[i].stakingToNodeId[msg.sender];
+      if (nodeId == address(0)) {
+        continue;
+      }
+      uint256 oldStakingBalance = trackers[i].stakingBalances[msg.sender];
+      uint256 newStakingBalance = msg.sender.balance;
+      trackers[i].stakingBalances[msg.sender] = newStakingBalance;
+      trackers[i].nodeBalances[nodeId] += (newStakingBalance - oldStakingBalance);
+
+      uint256 numGCs = trackers[i].nodeIds.length;
+      uint256 balance = trackers[i].nodeBalances[nodeId];
+      uint256 oldVotes = trackers[i].votingPowers[nodeId];
+      uint256 newVotes = min(numGCs - 1, balance / MIN_STAKE);
+      trackers[i].votingPowers[nodeId] = newVotes;
+      trackers[i].totalVotes += (newVotes - oldVotes);
+    }
+  }
+
+  // Called by CnStakingV2 to change voter account.
+  function appointVoter(address _voter) public {
+    for (uint i=0; i < liveTrackers.length; i++) {
+      if (block.number >= trackers[i].trackEnd) {
+        deleteArrayElem(liveTrackers, i);
+        continue;
+      }
+      address nodeId = trackers[i].stakingToNodeId[msg.sender];
+      if (nodeId == address(0)) {
+        continue;
+      }
+      uint256 oldVoter = trackers[i].nodeIdToVoter[nodeId];
+      uint256 newVoter = _voter;
+      trackers[i].voterToNodeId[oldVoter] = address(0);
+      trackers[i].voterToNodeId[newVoter] = nodeId;
+      trackers[i].nodeIdToVoter[nodeId] = newvoter;
+    }
+  }
+}
+```
+
+## Expected Effect
+The proposed GC Voting method is expected to produce the following changes:
+- All members in Klaytn ecosystem grow together with credibility 
+- Providing obligation and authority to GC motivates active participation in governance voting activities and the KLAY staking 
+- Anyone can easily view the proposals and voting status through the governance portal. It encourages holders to participate and give responsibility and authority to GCs.
+- The Klaytn network can take a step closer to transparency and decentralized networks.
+ 
+## Backward Compatibility
+- GCs must deploy new CnStakingV2 contracts and move their staked KLAYs. Existing CnStaking contract balances are excluded from voting power calculation.
```

Comment:
Is it okay to use old CnStaking contract if the GC does not want to have voting power?

Reply:
Yes, such GC would have 0 voting power but still can participate in block validation.
I've reworded must -> should.